project-icp/bee-pollinator-app
# Add Crop Data Layer

## Overview

Adds a button to toggle the crop layer. Toggling it on reveals an opacity slider that can be used to change the opacity of the layer. It defaults to 50%, same as in Pollination Mapper. It also reveals a legend of values. Unlike Pollination Mapper, the legend cannot be collapsed (this may be done as an enhancement). Clicking the layer button again closes the layer, but remembers the opacity for next time.

Also upgrades React Leaflet to 2, and adds React Leaflet Control.

Connects #367 

### TODO

- [x] Add Legend
- [x] Add Opacity Slider
- [x] Fix bug where CDL button is not rendered at start if not logged in

### Demo

~~https://user-images.githubusercontent.com/1430060/50489835-f2f0d980-09d7-11e9-90c8-699c1856a0d3.gif~~ Old Demo GIF

![2018-12-28 16 27 07](https://user-images.githubusercontent.com/1430060/50528671-86e8a100-0abd-11e9-9355-3b35cee66e8b.gif)

### Notes

~~There's a strange behavior where the CDL button won't render if there are no apiaries at first.~~ Fixed in 7782f5c.

## Testing Instructions

* Check out this branch, `beekeepers build` and go to [:8000/?beekeepers](http://localhost:8000/?beekeepers)
* Ensure you see the "Layers" button
* Ensure clicking it toggles the layer
* Ensure layer ordering is correct. It should be: basemap < CDL < apiaries < controls.
* Ensure sliding changes opacity
* Ensure legend matches the values in Pollination Mapper
* Run `./scripts/bundle.sh --debug` and go to [:8000/](http://localhost:8000/) and ensure the layer slider there still works.
